sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -165,21 +165,19 @@ namespace Azure { namespace Core {
      * within the context tree.
      *
      * @return If found, returns `true`, with \p outputValue set to the value associated with the
-     * key found; otherwise returns `false`.
+     * key found; otherwise returns `false`, including when the key found is not of the expected
+     * type.
      *
      * @remark The \p outputValue is left unmodified it the \p key is not found.
      */
     template <class T> bool TryGetValue(Key const& key, T& outputValue) const
     {
       for (auto ptr = m_contextSharedState; ptr; ptr = ptr->Parent)
       {
-        if (ptr->Key == key)
+        // Only return true if we find a key-type pair that matches.
```

Comment:
This is a contract violation as the code running after this point will NOT do what the customer expects.  I would expect this to be an abort().
Note: The keys are tied to a specific value instance.  They are not string keys.  Thus there should never be a key inserted that has an invalid type tied to it.

Reply:
Contexts with the same key can be added in the graph to override a previous value, there is no reason to block changing the type of that value as an override as well. So, this isn't really considered a contract violation. It should continue to work as customers expect.

Reply:
I agree you can add the same value to override.  I don't agree that changing the type of the object associated with that key is expected to work.  Their is a type coupling of the Key -> typeof(Value).  We need to help customers find their bugs when they try to put the wrong type in with a Key.